shared/localization/swap-locale.js

```diff
@@ -49,7 +49,9 @@ function swapLocale(lhr, requestedLocale) {
   // Copy LHR to avoid mutating provided LHR.
   lhr = JSON.parse(JSON.stringify(lhr));
 
-  const locale = i18n.lookupLocale(requestedLocale);
+  if (!format.hasLocale(requestedLocale)) {
```

Comment:
I think this is a behavior change, can we define it? Does this mean you have to pass in the exact locale name for this script to work, whereas before it could have been resolved to a supported locale via `lookupLocale`?

I don't think this has any implications, jw. #10148 will only ever pass a value that is one of the [locale .json files](https://github.com/GoogleChrome/lighthouse/pull/10148/files#diff-7b203d20c306c058b5f24385934941ca86bec075d746ad707dba58a0a765a2eeR55) (minus the `.json`), which is also always a keyof `LOCALES`. Admittedly it takes too many steps to see that this is correct.

Reply:
> I think this is a behavior change, can we define it? Does this mean you have to pass in the exact locale name for this script to work, whereas before it could have been resolved to a supported locale via `lookupLocale`?
> 
> I don't think this has any implications, jw. #10148 will only ever pass a value that is one of the [locale .json files](https://github.com/GoogleChrome/lighthouse/pull/10148/files#diff-7b203d20c306c058b5f24385934941ca86bec075d746ad707dba58a0a765a2eeR55) (minus the `.json`), which is also always a keyof `LOCALES`. Admittedly it takes too many steps to see that this is correct.

Yeah, so we know what we want locale fallback to act like in node, but it's not clear what it should do in the browser (e.g. the node version spends most of its time trying to figure out if `Intl` is enabled and telling you how to install a polyfill if not, which isn't ever an issue our supported browsers).

Since #10148 is going to include a list of our locales and one of the possible PSI designs is we automatically decide a locale and ship it in the bundle, it seems better to keep it simple here, include just what #10148 needs instead of speculating, and only add to it if we need a fallback mechanism in the future.

Reply:
Silently falling back to english in case of a non-existent locale request (like `i18n.lookupLocale` does) is good for locale lookup but doesn't seem ideal from `swap-locale`, as well.

Even with some fallback mechanism eventually added, if in the end no supported locale is found, `swap-locale` should probably complain loudly, whatever page is running it can absorb the error (show it to the end user if helpful), and then just not re-render the report